tsdb/head.go

```diff
@@ -1230,6 +1309,37 @@ func (a *headAppender) Append(ref uint64, lset labels.Labels, t int64, v float64
 	return s.ref, nil
 }
 
+func (a *headAppender) AppendExemplar(ref uint64, lset labels.Labels, e exemplar.Exemplar) (uint64, error) {
+	// Ensure no empty labels have gotten through.
```

Comment:
If I read the correctly, this code is not feature flagged (its the exemplar store that is, right?).  Therefore we need to be super careful here about performance.

In particular, we're doing the hash etc unconditionally.  We should follow the pattern in `Append` - lookup the reg first, and if that fails do the hash etc...

I'd also be comfortable failing AppendExemplar if the ref fails TBH - it should never create a series.

Reply:
`I'd also be comfortable failing AppendExemplar if the ref fails TBH - it should never create a series.` Yeah this makes sense to me as well.